src/crypto/sign/ed25519.rs

```diff
@@ -153,21 +153,21 @@ pub fn verify_detached(
 }
 
 /// Convert Ed25519 public key to Curve25519 public key.
-pub fn convert_ed_pk_to_curve25519(pk: PublicKey) -> [u8; SCALARMULTBYTES] {
-    let pk = clone_into_array(&pk[..]);
+pub fn convert_ed_pk_to_curve25519(PublicKey(ref pk): PublicKey) -> [u8; SCALARMULTBYTES] {
```

Comment:
Interesting, I rarely see such pattern. :upside_down_face: 

@ozkriff @vitvakatu

Reply:
I'm not sure that pattern matching in argument position is really justified in this case - `&pk.0` looks much less surprising for most Rust programmer. Do we have any code style agreement about this?

Reply:
Hm, this pattern was widely used in `sodiumoxide`, but now they prefer `pk.0`.

Reply:
I prefer `pk.0` too. :upside_down_face: